internal/customtypes/common_errors.go

```diff
@@ -0,0 +1,7 @@
+package customtypes
+
+import "errors"
+
+var (
+	ErrCustomTypeNil = errors.New("failed to set value. custom type is nil")
```

Comment:
Do you have a sample of what the output would look like when this error pops up? It seems like this verbiage might be confusing to a user, since "custom type" might not mean anything to them.